### ISSUE TITLE
[VHDL] unsupported entity port type 

### ISSUE THREAD

**Issue by Kelnorim** [doxygen/doxygen#7785]
**Describe the bug**
While generating documentation from VHDL sources few entities went missing.
I narrowed it down to the following port declaration : 
**_type vector_array is array (natural range <>) of std_logic_vector;_**
If I removed this port from the entity all the instances are properly documented.
Wondering if unconstrained 2D arrays as port type is a pending limitation ?

**Expected behavior**
All instances appear in the documentation

**To Reproduce**
declare an entity where one port type is : 
type vector_array is array (natural range <>) of std_logic_vector

**Version**
1.8.0

**Stack trace**
No crash nor warning on the failing entity.
However all components instantiated within the failing entity went also missing and the folowing warning is reported : 
**_warning: Internal inconsistency: member xxx_inst does not belong to any container!_**

**Additional context**
VHDL sources

**Comment by albert-github** [comment]
Doxygen version 1.8.0 is quite old (February 25, 2012), the current doxygen version is 1.8.18 and in the VHDL parsing a lot has changed

It is good to know the statement that causes the problem but to be able to reproduce:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.
- It would also be beneficial when you could try the current doxygen version 1.8.18).

**Comment by albert-github** [comment]
As a non-VHDL programmer it is a bit hard to debug , but I added the mentioned line:
```
type vector_array is array (natural range <>) of std_logic_vector;
```
to an existing vhdl file (including some documentation) and with the 1.8.0 version I see that some pages aren't generated (like html/classfilter__pkg.html) although doxygen says it finished properly.
When running the current doxygen version (1.9.3 (6518ff3d24ad187b7072bee854d69e285cd366ea)) I see that the page is generated.

I think the problem has been solved in the mean time.

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/7797570/example.tar.gz)
